### PR TITLE
Do not move last comment out of bounded expressions

### LINKED ISSUE #2608: Formatter shouldn't move a comment from the end of a bounded expression to the beginning of the following expression
```gleam
fn foo() {
  [1, 2,
    // abc
  ]

  case True {
    _ -> Nil
    // def
  }

  #(1, 2,
    // ghi
  )

  case True {
    _ -> #(
      1,
      2,
      // jkl
    )
  }
}
```

Currently formats to:

```gleam
fn foo() {
  [1, 2]

  // abc
  case True {
    _ -> Nil
  }

  // def
  #(1, 2)

  // ghi
  case True {
    _ -> #(1, 2)
  }
  // jkl
}
```

Not sure how it should be improved, but current behaviour is surely undesirable.

### PR DESCRIPTION
**Pull request by giacomocavalieri**
This PR closes #2608

### CODE REVIEW

File: compiler-core/src/format/tests.rs [hunk 1]
```diff
@@ -5549,3 +5549,40 @@ fn pipeline_inside_tuple_is_not_nested_if_only_item() {
 "#
     );
 }
+
+// github.com/gleam-lang/gleam/issues/2608
+#[test]
+fn comments_are_not_moved_out_of_bounded_expressions() {
```
Comment (lpil) [inline compiler-core/src/format/tests.rs:5555]: Could you make a test per thing being tested please 🙏

**Review by lpil — COMMENTED** [review]
Fab! One small comment below 🙏

File: compiler-core/src/format/tests.rs (hunk 1 above)
Comment (giacomocavalieri) [reply compiler-core/src/format/tests.rs:5555]: Sure!

File: compiler-core/src/format/tests.rs [hunk 2]
```diff
@@ -5549,3 +5549,67 @@ fn pipeline_inside_tuple_is_not_nested_if_only_item() {
...
+  ]
+
+  [
+    wibble,
+    wobble,
+    // list
+  ]
+}
+"#
+    );
```
Comment (lpil) [inline compiler-core/src/format/tests.rs:5570]: There's still two things being tested in this test!

Comment (giacomocavalieri) [reply compiler-core/src/format/tests.rs:5579]: Oops fixed it!

**Review by lpil — APPROVED** [review]
Thank you!

**Pull request merged**